mcf-browser/src/back-end/parse.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Series} from './data';
+import {ParseMcf} from './parse-mcf';
+import {ParseTmcf} from './parse-tmcf';
+import {ParsingError, ERROR_MESSAGES} from './utils';
+
+ type ParseFileResponse = {
+   /** A list of errors that occurred while parsing
+    * the files
+    */
+   errMsgs: ParsingError[];
+
+   /** A list of the ids of the subject nodes */
+   localNodes: string[];
+
+   /** A list of the generated mcf string for
+    * each CSV row
+    */
+   datapoints: Object[];
+ };
+
+/** Parse files and get the local nodes
+  * @param {Array<Blob>} fileList the list of blobs to be parsed
+  * @return {Object} an object containing all of the ids of the
+  *    subject nodes and the error messages
+  */
+async function getNodes(fileList: Blob[]) {
```

Comment:
please include the return type. Here and all other functions

Reply:
Added!